plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -16,10 +18,183 @@
  * @since 9.2.0
  * @package WooCommerce\Classes
  */
-class RemoteLogger {
+class RemoteLogger extends \WC_Log_Handler {
+	const LOG_ENDPOINT                = 'https://public-api.wordpress.com/rest/v1.1/logstash';
+	const RATE_LIMIT_ID               = 'woocommerce_remote_logging';
+	const RATE_LIMIT_DELAY            = 60; // 1 minute.
 	const WC_LATEST_VERSION_TRANSIENT = 'latest_woocommerce_version';
 	const FETCH_LATEST_VERSION_RETRY  = 'fetch_latest_woocommerce_version_retry';
 
+	/**
+	 * The logger instance.
+	 *
+	 * @var \WC_Logger_Interface|null
+	 */
+	private $local_logger;
+
+	/**
+	 * Remote logger constructor.
+	 *
+	 * @internal
+	 * @param \WC_Logger_Interface|null $logger Logger instance.
+	 */
+	public function __construct( \WC_Logger_Interface $logger = null ) {
+		if ( null === $logger ) {
+			$this->local_logger = wc_get_logger();
+		}
+		$this->local_logger = $logger;
```

Comment:
Apologies for missing this one during the earlier passes 😬 

I think though we need to make a change, else we'll often end up with the local logger set to null (and, we'll subsequently try to call a method on null, if for instance we try to use it to record *'Remote logging throttled').*

We probably need either:

```suggestion
		$this->local_logger = null === $logger
			? wc_get_logger()
			: $logger;
```

Or, if preferred:

```suggestion
		if ( null === $logger ) {
			$this->local_logger = wc_get_logger();
		} else {
			$this->local_logger = $logger;
		}
```

Reply:
Ah, Good catch!